### PR TITLE
Add in the SBT site mappings.

### PR DESCRIPTION
**Pull request by jkinkead**
Hah, I kept remembering this was missing, but kept forgetting to add it . . . :)

I also forgot to test the non-hacked version of the plugin on a real github pages.

### CODE REVIEW

File: src/main/scala/org/allenai/plugins/ScaladocGenPlugin.scala [hunk 1]
```diff
@@ -216,7 +216,10 @@ object ScaladocGenPlugin extends AutoPlugin {
         scalaApiMappingsSetting,
         // This will pull in links from pom.xml files that sbt has generated, when the project sets
         // the apiURL key.
-        autoAPIMappings := true
+        autoAPIMappings := true,
+        // This adds the output of Unidoc to the SbtSite plugin's site directory (what gets synced
+        // to github pages).
```
Comment (markschaake) [inline src/main/scala/org/allenai/plugins/ScaladocGenPlugin.scala]: Is this producing a site index or something? I don't know these terms so can't grok the comment here.

Comment (jkinkead) [inline src/main/scala/org/allenai/plugins/ScaladocGenPlugin.scala]: `sbt-site` just copies files into `target/site`, based on mappings. It's like the universal plugin's `stage` command.

`sbt-ghpages` then takes this directory and copies it to the `gh-pages` branch, then pushes it to github.

This is saying "take the mappings unidoc provides for the site plugin, and put the output in "latest/api"".

**Comment by jkinkead** [comment]
I tried to clarify the comment a bit.

**Comment by markschaake** [comment]
LGTM

**Pull request merged**